File: backend/app/domain/lane_pressure.py
"""Lane pressure domain models for map control tracking."""

from typing import Optional
from sqlmodel import SQLModel


class LanePressureSnapshot(SQLModel):
    """Snapshot of lane pressure at a given time.

    Pressure represents how far into enemy territory the wave has pushed.
    - 0.0 = wave at own base
    - 1.0 = wave at enemy base
    """

    pressure: float  # 0-1 value representing map pressure
    team: int  # Team (2 = Amber, 3 = Sapphire)
    attributed_players: list[int]  # Player custom_ids within proximity
    wave_x: float  # Wave centroid X position
    wave_y: float  # Wave centroid Y position
    wave_count: int  # Number of creeps in the wave


class LanePressureData(SQLModel):
    """Container for all lane pressure data.

    Key format: "{lane}_{team}" (e.g., "1_2" for lane 1, team 2/Amber)
    Value: per-second snapshots (None if no wave that second)
    """

    pressure: dict[str, list[Optional[LanePressureSnapshot]]]

File: frontend/src/domain/lanePressure.ts
// Lane pressure types for map control tracking

export interface LanePressureSnapshot {
  pressure: number; // 0-1 value (0 = own base, 1 = enemy base)
  team: number; // Team (2 = Amber, 3 = Sapphire)
  attributed_players: number[]; // Player custom_ids within proximity
  wave_x: number; // Wave centroid X position
  wave_y: number; // Wave centroid Y position
  wave_count: number; // Number of creeps in the wave
}

// Key format: "{lane}_{team}" (e.g., "1_2" for lane 1, team 2/Amber)
// Value: per-second snapshots (null if no wave that second)
export interface LanePressureData {
  pressure: Record<string, (LanePressureSnapshot | null)[]>;
}
